File: src/main/java/com/tatsuaki/carepreventioncsv/model/domain/CarePreventionCsvColumn/ConsignmentDivisionInsuranceAssociation.java
package com.tatsuaki.carepreventioncsv.model.domain.CarePreventionCsvColumn;

import java.util.Arrays;
import java.util.List;

import static java.util.stream.Collectors.toList;

public class ConsignmentDivisionInsuranceAssociation {

    private final String consignmentDivisionInsuranceAssociation;

    public enum Code {

        TARGET("1", "委託する"),
        NON_TARGET("2", "委託しない");

        private String code;
        private String japanese;

        Code(String code, String japanese) {
            this.code = code;
            this.japanese = japanese;
        }

        public String getCode() {
            return code;
        }

        public String getJapanese() {
            return japanese;
        }

        public static boolean contains(String targetCode) {
            List<ConsignmentDivisionInsuranceAssociation.Code> targetCodes = Arrays.stream(values())
                    .filter( value -> value.getCode().equals(targetCode) )
                    .limit(1)
                    .collect(toList());

            return targetCodes.size() != 0;
        }

        public static String getJapanese(String targetCode) {
            List<ConsignmentDivisionInsuranceAssociation.Code> targetCodes = Arrays.stream(values())
                    .filter( value -> value.getCode().equals(targetCode) )
                    .limit(1)
                    .collect(toList());

            if (targetCodes.size() == 0) {
                return "";
            }

            return targetCodes.get(0).getJapanese();
        }
    }

    public ConsignmentDivisionInsuranceAssociation(String consignmentDivisionInsuranceAssociation) {
        this.consignmentDivisionInsuranceAssociation = consignmentDivisionInsuranceAssociation;
    }

    public String getConsignmentDivisionInsuranceAssociation() {
        return consignmentDivisionInsuranceAssociation + getJapanese(consignmentDivisionInsuranceAssociation);
    }

    private String getJapanese(String consignmentDivisionInsuranceAssociation) {
        return validateCodeFormat() ?
                "(" + ConsignmentDivisionInsuranceAssociation.Code.getJapanese(consignmentDivisionInsuranceAssociation) + ")" : "";
    }

    public String getFormatErrorMessage() {
        StringBuilder errorMessageBuilder = new StringBuilder();

        if (!validateCodeFormat()) {
            errorMessageBuilder.append("存在しない事業対象者実施区分コードです.");
        }

        return errorMessageBuilder.toString();
    }

    private boolean validateCodeFormat() {
        return ConsignmentDivisionInsuranceAssociation.Code.contains(consignmentDivisionInsuranceAssociation);
    }
}

File: src/main/kotlin/com/tatsuaki/carepreventioncsv/model/domain/csvcolumn/ConsignmentDivisionInsuranceAssociation.kt
package com.tatsuaki.carepreventioncsv.model.domain.csvcolumn

import java.util.Arrays

import java.util.stream.Collectors.toList

/**
 * 事業対象者実施区分コード
 */
class ConsignmentDivisionInsuranceAssociation(consignmentDivisionInsuranceAssociation: String) : CsvColumn(consignmentDivisionInsuranceAssociation) {

    override fun getFormatErrorMessage(): String {
        val errorMessageBuilder = StringBuilder()

        if (!validateCodeFormat()) {
            errorMessageBuilder.append("存在しない事業対象者実施区分コードです.")
        }

        return errorMessageBuilder.toString()
    }

    val description: String
        get() {
            return content + getJapanese(content)
        }

    private enum class Code constructor(val code: String, val japanese: String) {

        TARGET("1", "委託する"),
        NON_TARGET("2", "委託しない");

        companion object {

            fun contains(targetCode: String): Boolean {
                val targetCodes = Arrays.stream(values())
                        .filter { value -> value.code == targetCode }
                        .limit(1)
                        .collect(toList())

                return targetCodes.size != 0
            }

            fun of(targetCode: String): Code {
                val targetCodes = Arrays.stream(values())
                        .filter { value -> value.code == targetCode }
                        .limit(1)
                        .findFirst()
                        .get()

                return targetCodes
            }
        }
    }

    private fun getJapanese(consignmentDivisionInsuranceAssociation: String): String {
        return if (validateCodeFormat())
            "(" + Code.of(consignmentDivisionInsuranceAssociation).japanese + ")"
        else
            ""
    }

    override fun getFormatContent(): String {
        return content + getJapanese(content)
    }

    override fun validateCodeFormat(): Boolean {
        return Code.contains(content)
    }

    override fun validateCharacter(): Boolean {
        return true
    }

    override fun validateLength(): Boolean {
        return true
    }
}
